llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -10519,8 +10521,10 @@ BoUpSLP::getEntryCost(const TreeEntry *E, ArrayRef<Value *> VectorizedVals,
     auto GetVectorCost = [&](InstructionCost CommonCost) {
       auto *MaskTy = getWidenedType(Builder.getInt1Ty(), VL.size());
 
-      InstructionCost VecCost = TTI->getCmpSelInstrCost(
-          E->getOpcode(), VecTy, MaskTy, VecPred, CostKind, VL0);
+      InstructionCost VecCost =
+          TTI->getCmpSelInstrCost(E->getOpcode(), VecTy, MaskTy, VecPred,
+                                  CostKind, getOperandInfo(E->getOperand(0)),
+                                  getOperandInfo(E->getOperand(1)), VL0);
```

Comment:
Maybe use generic data here too to add full support later along with the tests?

Reply:
Can you rephrase?  I can't figure out your intent.  My best guess is that you want me to split the patch into an SLP portion and a TTI portion?  Is that what you meant?

Reply:
Kind of. Use default values here and use actual operand info in a separate patch

Reply:
Done.